darion/api/handlers_test.go

```diff
@@ -168,7 +215,7 @@ func verifyHelper(resp *http.Response, t *testing.T) string {
 		t.Fatal("Could not read response body: ", err)
 	}
 	dataStr := string(data)
-	if resp.StatusCode != 200 {
```

Comment:
When will we receive not 200? When it's range request not satisfiable?

Reply:
If that's the case, we should probably explicitly check for that.

Reply:
If a range is requested http specs say the server should respond with a  206 "Partial Content".

In general though, any code between 200 and 399 is considered a success. And success checks should allow for anything in that range. For example 304s would be pretty common. But I can change the test to only allow 200 or 206.

Reply:
BTW the code for unsatisfiable range requests is 416.

Reply:
Do we want add a specific test checking for the 416 case, and for the 206 case?

Reply:
Sure, added. Though at that point we are testing the http lib not our code.